src/main/java/athena/TaskList.java

```diff
@@ -217,10 +218,14 @@ public TaskList getFilteredList(TaskFilter taskFilter) {
             if (taskFilter.isTaskIncluded(task)) {
                 if (taskFilter instanceof ImportanceFilter) {
                     filteredTasks.add(task);
-                } else {
+                } else if (taskFilter instanceof ForecastFilter) {
                     assert taskFilter instanceof ForecastFilter;
                     Task filteredTask = ((ForecastFilter) taskFilter).removeExcludedDates(task);
                     filteredTasks.add(filteredTask);
+                } else {
+                    assert taskFilter instanceof FlexibleTimeFilter;
+                    filteredTasks.add(task);
+
```

Comment:
How about

```
if (taskFilter instanceof ForecastFilter) {
    ...
}
else {
    filteredTasks.add(task);
}
```

since the other filters don't need to do anything.

Reply:
That works too